tools/generate_csv/generate_csv.py

```diff
@@ -96,128 +184,220 @@ def __init__(self, url, qidForDebuggingOnly):
-csv_file.write("pictureLicenseURL;")
-csv_file.write("type;")
-csv_file.write("typeQID;")
-csv_file.write("creation;")
-csv_file.write("QID;")
-csv_file.write("\n")
-
-#print(get_pois_for_operator("http://www.wikidata.org/entity/Q242"))
-operators = get_operators()
-for operator in operators:
-    print("=== operator ===")
-    print(operator)
-    operator_label = operator.get("operatorLabel").get("value")
-    print("=== operator_label ===")
-    print(operator_label)
-    operator_qid = operator.get("operator").get("value")
-    print("=== operator_qid ===")
-    print(operator_qid)
-    csv_file.write(get_pois_for_operator(operator_label, operator_qid))
+
+def write_tmp_csv(thread_id, operators):
+<<<<<<< HEAD
+    csv_file = open(f"tmp/database_of_embassies_{thread_id}.csv", "w")
+=======
+    csv_file = open(f"tmp/database_of_embassies_{thread_id}.csv", 'w')
+>>>>>>> 84564713104e9f3d6c867acef80ccd69b0c0e567
+    for operator in operators:
+        print("=== operator ===")
+        print(operator)
+        operator_label = operator.get("operatorLabel").get("value")
+        print("=== operator_label ===")
+        print(operator_label)
+        operator_qid = operator.get("operator").get("value")
+        print("=== operator_qid ===")
+        print(operator_qid)
+        csv_file.write(get_pois_for_operator(sparql, operator_label, operator_qid))
+
+<<<<<<< HEAD
+
+def get_sublists(lst: List[str], n: int) -> List[List]:
+    subListLength = len(lst) // n
+    return [lst[i : i + subListLength] for i in range(0, len(lst), subListLength)]
+
+
+if __name__ == "__main__":
+=======
+def get_sublists(lst: List[str],n: int) -> List[List]:
+    subListLength = len(lst) // n 
+    return [lst[i:i + subListLength] for i in range(0, len(lst), subListLength)]
+
+if __name__ == "__main__":
+    makedirs("tmp", exist_ok=True)
+>>>>>>> 84564713104e9f3d6c867acef80ccd69b0c0e567
+    sparql_endpoint_url = "https://query.wikidata.org/sparql"
+    sparql = SPARQLWrapper(sparql_endpoint_url, agent=user_agent)
+    sparql.setReturnFormat(JSON)
+
+<<<<<<< HEAD
+    with open("pois_for_operator.sparql", "r") as file:
+        query_template = file.read()
+
+    csv_file = open("database_of_embassies.csv", "w")
+    columns_name = [
+        "operator",
+        "operatorQID",
+        "jurisdictions",
+        "jurisdictionQIDs",
+        "country",
+        "countryQID",
+        "city",
+        "cityQID",
+        "latitude",
+        "longitude",
+        "phone",
+        "email",
+        "website",
+        "facebook",
+        "twitter",
+        "youtube",
+        "picture",
+        "pictureAuthor",
+        "pictureLicense",
+        "pictureLicenseURL",
+        "type",
+        "typeQID",
+        "creation",
+        "QID",
+    ]
+    csv_file.write(f"{';'.join(columns_name)}\n")
+
+    # print(get_pois_for_operator("http://www.wikidata.org/entity/Q242"))
+    operators = get_operators(sparql)
+    number_of_thread = 20
```

Comment:
In the past I have brought the Wikidata server to its knees so I prefer to be prudent...
If you want to make the script a bit faster, wouldn't it be easier to just reduce `time.sleep` a bit?

Reply:
yes, I assumed that few parallel operations will be able to not overload the server will keeping the sleep you put in place, I can move this out of PR and keep it in the fork if you don't need it

Reply:
Having diverging forks is not in our mutual interest. How about having a default value of `3` instead of `20`?

Very minor: How about renaming `number_of_thread` to `number_of_threads`?